pages/_app.js

```diff
@@ -17,6 +23,33 @@ export default class MyApp extends App {
 		return { pageProps }
 	}
 
+	componentDidUpdate() {
+		if (this.state.currentRoute !== this.props.router.route) {
+			this.setState(
+				{
+					currentRoute: this.props.router.route
+				},
+				() => {
+					analyticsService.trackPage(this.state.currentRoute)
+				}
+			)
+		}
+	}
+
+	/**
+	 * Initializes GA if has not been initialized yet
+	 */
+	componentDidMount() {
```

Comment:
Stores the current route/path in React state and does a check to see if we need to initialize GA